pkg/sdk/testint/secrets_gen_integration_test.go

```diff
@@ -0,0 +1,673 @@
+		require.NoError(t, err)
+		require.Contains(t, returnedSecrets, *secret)
+
+		returnedSecrets, err = client.Secrets.Show(ctx, sdk.NewShowSecretRequest().WithIn(sdk.In{Schema: id.SchemaId()}))
+		require.NoError(t, err)
+		require.Contains(t, returnedSecrets, *secret)
+	})
+
+	t.Run("Show: with In", func(t *testing.T) {
+		secretOAuthClientCredentials, id := createSecretWithOAuthClientCredentialsFlow(t, integrationId, []sdk.SecurityIntegrationScope{{Scope: "foo"}}, nil)
+		secretOAuthAuthorizationCode, _ := createSecretWithOAuthAuthorizationCodeFlow(t, integrationId, "foo", refreshTokenExpiryTime, nil)
+		secretBasicAuthentication, _ := createSecretWithBasicAuthentication(t, "foo", "bar", nil)
+		secretGenericString, _ := createSecretWithGenericString(t, "foo", nil)
+
+		returnedSecrets, err := client.Secrets.Show(ctx, sdk.NewShowSecretRequest().WithIn(sdk.In{Account: sdk.Pointer(true)}))
+		require.NoError(t, err)
+		require.Contains(t, returnedSecrets, *secretOAuthClientCredentials)
+		require.Contains(t, returnedSecrets, *secretOAuthAuthorizationCode)
+		require.Contains(t, returnedSecrets, *secretBasicAuthentication)
+		require.Contains(t, returnedSecrets, *secretGenericString)
+
+		returnedSecrets, err = client.Secrets.Show(ctx, sdk.NewShowSecretRequest().WithIn(sdk.In{Database: id.DatabaseId()}))
+		require.NoError(t, err)
+		require.Contains(t, returnedSecrets, *secretOAuthClientCredentials)
+		require.Contains(t, returnedSecrets, *secretOAuthAuthorizationCode)
+		require.Contains(t, returnedSecrets, *secretBasicAuthentication)
+		require.Contains(t, returnedSecrets, *secretGenericString)
+
+		returnedSecrets, err = client.Secrets.Show(ctx, sdk.NewShowSecretRequest().WithIn(sdk.In{Schema: id.SchemaId()}))
+		require.NoError(t, err)
+		require.Contains(t, returnedSecrets, *secretOAuthClientCredentials)
+		require.Contains(t, returnedSecrets, *secretOAuthAuthorizationCode)
+		require.Contains(t, returnedSecrets, *secretBasicAuthentication)
+		require.Contains(t, returnedSecrets, *secretGenericString)
+	})
+
+	t.Run("Show: SecretWithGenericString with In", func(t *testing.T) {
+		secret, id := createSecretWithGenericString(t, "foo", nil)
+
+		returnedSecrets, err := client.Secrets.Show(ctx, sdk.NewShowSecretRequest().WithIn(sdk.In{Account: sdk.Pointer(true)}))
+		require.NoError(t, err)
+		require.Contains(t, returnedSecrets, *secret)
+
+		returnedSecrets, err = client.Secrets.Show(ctx, sdk.NewShowSecretRequest().WithIn(sdk.In{Database: id.DatabaseId()}))
+		require.NoError(t, err)
+		require.Contains(t, returnedSecrets, *secret)
+
+		returnedSecrets, err = client.Secrets.Show(ctx, sdk.NewShowSecretRequest().WithIn(sdk.In{Schema: id.SchemaId()}))
+		require.NoError(t, err)
+		require.Contains(t, returnedSecrets, *secret)
+	})
+
+	t.Run("ShowByID", func(t *testing.T) {
+		_, id := createSecretWithGenericString(t, "foo", nil)
```

Comment:
To really test ShowById the best way would be to create a new schema on top of test database (you can do it with `	testClientHelper().Schema.CreateSchemaInDatabase(t, testClientHelper().Ids.DatabaseId())`) and then create a secret with the same name as the first one in this newly created schema, and then see what ShowById returns. Take a look at this test `"show by id - same name in different schemas",` as a good example.

Reply:
The test seems similar to what you have at the bottom of this file, but not quite.